api/internal/handler/route/route.go

```diff
@@ -489,12 +503,32 @@ func (h *Handler) Update(c droplet.Context) (interface{}, error) {
 		}
 	}
 
-	ret, err := h.routeStore.Update(c.Context(), &input.Route, true)
+	// check name existed
+	ret, err := h.routeStore.List(c.Context(), store.ListInput{
```

Comment:
All these logics are similar, can we abstract them to a common interface/method, so that we can reduce the code complexity and redundency.

Reply:
Good idea.

Reply:
We can use reflection to get the name filed.

Reply:
updated. thanks.

Reply:
@tokers @starsz  please review again when you have time.